From 4fb6d4208cfb20bad4a3491a18e78409b5a8183f Mon Sep 17 00:00:00 2001
From: Matthew Smith <matt@offtopica.uk>
Date: Sat, 30 Apr 2022 07:42:50 +0100
Subject: [PATCH] Don't invoke Python to create libexec/mold/ld symlink

Reverts commits 8073a92614fb59f59570031badab5dd4bc3b4f7f and
5803c3c200f301adc3abdb66df16d3d669712d70.

Bug #841575
---
 Makefile | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index d1abc850..10e053a9 100644
--- a/Makefile
+++ b/Makefile
@@ -204,12 +204,7 @@ install: all
 	$(STRIP) $D$(LIBDIR)/mold/mold-wrapper.so
 
 	$(INSTALL) -d $D$(LIBEXECDIR)/mold
-
-# We want to make a symblink with a relative path, so that users can
-# move the entire directory to other place without breaking the reference.
-# GNU ln supports `--relative` to do that, but that's not supported by
-# non-GNU systems. So we use Python to compute a relative path.
-	ln -sf `python3 -c "import os.path; print(os.path.relpath('$(BINDIR)/mold', '$(LIBEXECDIR)/mold'))"` $D$(LIBEXECDIR)/mold/ld
+	ln -sf $(BINDIR)/mold $D$(LIBEXECDIR)/mold/ld
 
 	$(INSTALL) -d $D$(MANDIR)/man1
 	$(INSTALL_DATA) docs/mold.1 $D$(MANDIR)/man1
-- 
2.35.3

